### PR TITLE
Add default settings route and make analysis settings required [TG-10029, TG-10032]

### PR DESCRIPTION
**Pull request by adamkowalczyk**
### Context/purpose

The client should support the new `/default-settings` api route.

The client should understand that settings merging has been removed and that settings are now required to start an analysis.

### Implementation details

Add low level and OO methods for `getDefaultSettings`

Make `settings` a required parameter for the low level start analysis binding.

Settings remain optional for the OO `start` and `run` methods, and the default settings will be fetched and applied to the analysis if the settings are omitted.

Various type and docs updates, including removing references to `ignoreDefaults`.

### QA instructions

You should now be able to get default settings from the API.

You should be able to start an analysis with default settings by not providing settings when using `Analysis.run` or `Analysis.start`.

You should no longer be able to call the low level `startAnalysis` function without providing settings.

### PR Checklist

- [x] I have given the PR a meaningful title which can be understood out of context (used in the changelog)
- [x] I have read through my code and fixed the indentation, followed naming conventions and removed leftover debugging
- [x] I have added the Jira ticket number(s) to the title with the format "Description of the feature [TG-123, TG-456]" and have copied the QA instructions to the Jira ticket (if one exists)

### CODE REVIEW

File: tests/unit/bindings.ts [hunk 1]
```diff
@@ -44,6 +45,28 @@ describe('api/bindings', () => {
...
+
+      const response = await getDefaultSettings(api);
+
+      assert.deepStrictEqual(response, settings);
+    }));
+
+    it('Handles the allowUnauthorizedHttps option correctly', sinonTest(async (sinon) => {
+      const get = sinon.stub(dependencies.request, 'get').resolves();
+      await getDefaultSettings(api, { allowUnauthorizedHttps: true });
+      assert.calledOnceWith(get, [defaultSettingsUrl, sampleConfig]);
```
Comment (JimmehAH) [inline tests/unit/bindings.ts]: I'm not sure what this is actually testing. Just that adding that option doesn't make things blow up?

Comment (adamkowalczyk) [reply tests/unit/bindings.ts]: It is testing that if bindings options are set on the `Analysis` object that they are passed to `bindings.getDefaultSettings` when it is called. This test exists for the other call throughs in methods.

File: docs/programmatic-interface.md [hunk 2]
```diff
@@ -125,6 +131,12 @@ Including `baseBuild` will enable a differential analysis.
 
 The second parameter is an optional settings object, containing analysis settings to be uploaded to the Diffblue Cover API.
 
+If the settings parameter is omitted, default settings will be fetched from the Diffblue Cover API (if they have not already been set on the `Analysis` object) and used to run the analysis.
+
+After calling `Analysis.start` the `settings`, `analysisId`, `computedSettings` and `status` properties of the analysis object will be updated
```
Comment (myron836) [inline docs/programmatic-interface.md]:
```suggestion
After calling `Analysis.start` the `settings`, `analysisId`, `computedSettings` and `status` properties of the analysis object will be updated.
```

File: docs/programmatic-interface.md [hunk 3]
```diff
@@ -125,6 +131,12 @@ Including `baseBuild` will enable a differential analysis.
...
+The `settings` property will contain the settings provided when calling `Analysis.start`, the `computedSettings` property will contain the settings
+used to start the analysis as returned from the server in the start analysis response.
```
Comment (myron836) [inline docs/programmatic-interface.md]: Not sure a newline is necessary here

File: docs/programmatic-interface.md [hunk 4]
```diff
@@ -194,10 +214,26 @@ To cancel an analysis that has started, call `Analysis.cancel`.
...
+
+To get a set of default recommended analysis settings from the Diffblue Cover API, call `Analysis.getDefaultSettings`.
+
+The returned default settings will be stored in the `defaultSettings` property of the analysis object.
+
+```ts
+(async () => {
+ const defaultSettings = await analysis.getDefaultSettings();
+ console.log('Default analysis settings:');
+ console.dir(defaultSettings)
```
Comment (myron836) [inline docs/programmatic-interface.md]:
```suggestion
 console.dir(defaultSettings);
```

**Comment by adamkowalczyk** [comment]
> Not sure how I feel about the low level and high level bindings having different behaviour, why make one require you to get settings whereas the other not?

My thinking was that the binding is for making a direct call to the API, and the start analysis endpoint requires that you provide a settings object, so the binding should as well.

The `Analysis` class is designed to help people with running an analysis, it does more than just call through to the API bindings, so I decided to add the helpful feature of allowing people to omit the settings and have the class fetch the defaults.

I feel it is analogous to the way that a user would need to do cursor pagination of results themselves when using the bindings, but the `Analysis` class handles it automatically.

**Pull request merged**